### PR TITLE
CLI integration test beginnings

### PR DESCRIPTION
**Pull request by yaleman**
Things!

* Made the token cache path configurable at runtime, it's a hidden option but necessary so testers don't stomp their local token cache
* Added some real "run the actual tests using the server and CLI" tests
* Tweaked the logging in test mode so `escargot` (which builds the above CLI runner) and `hyper` are less noisy in tests.
* Added an automagical way to print output based on selected CLI output mode (OutputMode::print_message).
* Clippy was sad, made the crab happy.

Checklist

- [x] This pr contains no AI generated code
- [x] cargo fmt has been run
- [x] cargo clippy has been run
- [x] cargo test has been run and passes

### CODE REVIEW

File: book/src/integrations/pam_and_nsswitch.md [hunk 1]
```diff
@@ -183,6 +183,8 @@ cp -a /etc/pam.d /root/pam.d.backup
 
 Documentation examples for the following Linux distributions are available:
 
-* [Fedora](pam_and_nsswitch/fedora.md)
-* [SUSE / OpenSUSE](pam_and_nsswitch/suse.md)
-* Debian / Ubuntu - when one generates packages [from the repository tools](https://github.com/kanidm/kanidm/tree/master/platform/debian), configuration is modified on install.
+- [Fedora](pam_and_nsswitch/fedora.md)
+- [SUSE / OpenSUSE](pam_and_nsswitch/suse.md)
```
Comment (Firstyear) [inline book/src/integrations/pam_and_nsswitch.md:187]: You know the rule.

File: server/testkit/Cargo.toml [hunk 2]
```diff
@@ -23,27 +23,35 @@ default = []
 webdriver = ["fantoccini"]
 
 [dependencies]
+assert_cmd = "2.0.12"
+escargot = "0.5.8"
+# used for webdriver testing
+fantoccini = { version = "0.19.3", optional = true }
```
Comment (Firstyear) [inline server/testkit/Cargo.toml:29]: If it's for testing, could this be in dev-dependencies.

File: server/testkit/tests/integration.rs [hunk 3]
```diff
@@ -219,3 +227,135 @@ async fn test_idm_domain_set_ldap_basedn(rsclient: KanidmClient) {
...
+        .env("KANIDM_PASSWORD", IDM_ADMIN_TEST_PASSWORD)
+        .args(cmd.split(" "))
+        .output()
+        .unwrap();
+    println!("##############################\n{} result: {:?}", cmd, res);
+    assert!(res.status.success());
+    res
+}
+
+#[kanidmd_testkit::test(threads = 4)]
```
Comment (Firstyear) [inline server/testkit/tests/integration.rs:269]: Should we be hardcoding this value?

**Review by Firstyear — APPROVED** [review]
Some minor changes requested.

File: server/testkit/tests/integration.rs (hunk 3 above)
Comment (yaleman) [reply server/testkit/tests/integration.rs:269]: Lemme try removing it, I know I had to set it at one point because something blew up

Comment (yaleman) [reply server/testkit/tests/integration.rs:269]: yep, all gud

File: server/testkit/Cargo.toml (hunk 2 above)
Comment (yaleman) [reply server/testkit/Cargo.toml]: Will try that, I think it was being a jerk about things at one point.

**Pull request merged**